lib/functions/modem.js

```diff
@@ -700,49 +772,121 @@ module.exports = function (SerialPort) {
   }
 
   modem.getOwnNumber = (callback, timeout = 10000) => {
+    if (callback == undefined) {
+      return new Promise((resolve, reject) => {
+        modem.getOwnNumber((result, error) => {
+          if (error) {
+            reject(error)
+          } else {
+            resolve(result)
+          }
+        }, timeout)
+      })
+    }
     modem.executeCommand(`AT+CNUM`, result => {
       callback(result)
     }, false, timeout)
   }
 
   modem.setOwnNumber = (number, callback, name = 'OwnNumber', timeout = 10000) => {
+    if (callback == undefined) {
+      return new Promise((resolve, reject) => {
+        modem.setOwnNumber(number, (result, error) => {
+          if (error) {
+            reject(error)
+          } else {
+            resolve(result)
+          }
+        }, name, timeout)
+      })
+    }
     //Select Phonebook Memery Storage "ON" SIM (or ME) own numbers (MSISDNs) list (reading of this storage may be available through +CNUM
     modem.executeCommand(`AT+CPBS="ON"`, result => {
       if (result.data === 'OK') {
         //CPBW parameters, Phonebook location, phone number, type of address octet in integer format, name
         modem.executeCommand(`AT+CPBW=1,"${number}",129,"${name}"`, result => {
-          if (callback) callback(result)
```

Comment:
there will be always a callback provided, whether by caller or by promisifing block up here.